### PR TITLE
Mepts 1054 mq cat 14 indicator 15 16 den num mds

### PR DESCRIPTION
**Pull request by ningosi**
This PR checks for the MDS per the MQ and MI CAT 15.16, only the FICHA and FILA have been tested to have patients in the database, also on checking the patients having more than 3 months on ART base don VL, was missing encounter type 51.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java [hunk 1]
```diff
@@ -2075,4 +2076,93 @@ private static String getMetadataFrom(List<Integer> dispensationTypes) {
...
+            + " e.encounter_datetime <=:endDate AND "
+            + " e.location_id=:location "
+            + " UNION "
+            + " SELECT p.patient_id,o.obs_datetime AS encounter_date FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN "
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (${53}) AND o.concept_id=${856} AND o.value_numeric IS NOT NULL AND "
+            + " o.obs_datetime <=:endDate AND "
+            + " e.location_id=:location "
+            + " ) max_vl_date GROUP BY patient_id "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]: @ningosi can you please format the code

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 2]
```diff
@@ -226,4 +226,68 @@ public static String getPatientAgeBasedOnFirstViralLoadDate(
...
+            + " SELECT p.patient_id,o.obs_datetime AS encounter_date FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN "
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (%d) AND o.concept_id=%d AND o.value_numeric IS NOT NULL AND "
+            + " o.obs_datetime <=:endDate AND "
+            + " e.location_id=:location "
+            + " ) max_vl_date GROUP BY patient_id "
+            + ") vl_date_tbl ON pp.patient_id=vl_date_tbl.patient_id WHERE ee.encounter_datetime BETWEEN date_add(vl_max_date, interval -12 MONTH) AND date_add(vl_max_date, interval -1 DAY) AND ee.encounter_type=%d "
+            + " ) fin_tbl GROUP BY patient_id "
+            + ") out_p ON pp.patient_id=out_p.patient_id AND op.concept_id =%d AND op.value_coded IN(%d,%d) AND ep.encounter_type=%d AND ep.encounter_datetime=max_vl_date_and_max_ficha";
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java]: @ningosi format here also, just put it here https://www.dpriver.com/pp/sqlformat.htm

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
Please format the code

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 3]
```diff
@@ -324,20 +324,108 @@ public static String getPatientsHavingTypeOfDispensationBasedOnTheirLastVlResult
...
+        + "															AND e.voided = 0 "
+        + "															AND o.voided = 0 "
+        + "															AND e.encounter_type IN ( "
+        + "																13, 6, 9, 51 ) "
+        + "															AND "
+        + "															( ( o.concept_id = 856 "
+        + "															AND o.value_numeric IS NOT "
+        + "															 NULL ) "
+        + "															OR ( o.concept_id = 1305 "
+        + "															 AND o.value_coded IS NOT "
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java]: Maybe we need those substitutions to point to references using the ${}

**Comment by ningosi** [comment]
Hi @psmatsinhe you requested some changes on this PR, were they addressed? if Yes, accept them to give me a second chance to review

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**